### ISSUE TITLE
Doesn't work with \usepackage[cmyk]{xcolor}

### ISSUE THREAD

**Issue by dcpurton** [jbezos/colorspace#1]
See https://tex.stackexchange.com/q/435909/87678 for the question that precipitates this bug report.

Spot colours do not work if `xcolor` is loaded with the `cmyk` option. When you try and use a spot colour under these circumstances an error is produce:

```
! Argument of \c@lor@@cmyk has an extra }.
<inserted text> 
                \par 
l.9 \textcolor{registration}{Registration colour}
```

# MWE

In this MWE, my test colour is RGB red and CMYK cyan, so you easily see which colour is in use.

```TeX
\documentclass{article}
\usepackage[cmyk]{xcolor}
\usepackage{colorspace}
\definecolor{testcolour}{RGB/cmyk}{255,0,0/1,0,0,0}
\definespotcolor{registration}{All}{1,1,1,1}
\begin{document}
\textcolor{testcolour}{Test Colour}

\textcolor{registration}{Registration colour}
\end{document}
```

**Comment by jbezos** [comment]
Yes, the package option `cmyk` just calls `\selectcolormodel`, which, as explained in the manual, doesn't work yet. This is a limitation, which means `cmyk` must be selected explicitly when defining colors.

**Comment by jbezos** [comment]
Sadly, this is far from trivial. If, say, the color model `cmyk` is selected, `xcolor` assumes the value to be written to the PDF is _always_ `cmyk` (it forces it, and even `gray` is converted). Changing this behavior requires patching a good deal of internal macros. For the moment, it's a limitation we have to live with.

**Comment by dcpurton** [comment]
I think it's OK. Once I figured out that as long as the current colour model is `natural` when a spot colour is actually used it's fine. On reflection, I think that `\usepackage[cmyk]{xcolor}` should have the effect of outputting spot colours in CMYK anyway (perhaps this is easier to implement?), which is different to what I was initially thinking. (I initially expected that it would output everything in CMYK+spot.)

**Comment by jbezos** [comment]
Outputting spot colours in CMYK is also difficult (I considered this possibility, too), at least if done at the time the PDF is output.
